.github/workflows/tests.yml

```diff
@@ -35,12 +35,54 @@ jobs:
       - name: Coverage reporting
         run: |
           coverage combine
-          coverage report -m
-          coverage xml
-          coverage html
-      - name: Publish coverage results
-        uses: codecov/codecov-action@v4
+          coverage report --show-missing
+          mv .coverage .coverage.${{ matrix.python-version }}
```

Comment:
Usually, I tend to use the [`COVERAGE_FILE`](https://coverage.readthedocs.io/en/latest/cmd.html#data-file) env var to control the name of the coverage file, but it changes the way `coverage combine` works, and I was afraid I'd have to change more things for this to work, so I opted to move myself the .coverage into its own name.